stacks/golang/setup.sh

```diff
@@ -5,17 +5,22 @@
 
 set -euo pipefail
 
-version=1.19.2
-os=linux
-arch=amd64
-
 # The version of golang in the debian repositories tends to be incredibly
 # out of date; let's get ourselves a newer version from upstream:
-curl -L https://golang.org/dl/go${version}.${os}-${arch}.tar.gz -o go.tar.gz
+if [ -e /opt/app/.sandstorm/go-version ]; then
+    # Get the same version we've used before
+    curl -L "https://go.dev/dl/$(cat '/opt/app/.sandstorm/go-version').linux-amd64.tar.gz" -o go.tar.gz
+else
+    # Get the newest version for a new project
+    curl -L "https://go.dev/dl/$(curl 'https://go.dev/VERSION?m=text').linux-amd64.tar.gz" -o go.tar.gz
+fi
 tar -C /usr/local -xzf go.tar.gz
 rm go.tar.gz
 echo 'export PATH=/usr/local/go/bin:$PATH' > /etc/profile.d/go.sh
 
+# Get the same version next time
+/usr/local/go/bin/go version | cut -d ' ' -f 3 > /opt/app/.sandstorm/go-version
```

Comment:
If you wanted to avoid having to specify the full path, you could first `source /etc/profile.d/go.sh`, to make the change there affect the current shell. This is fine too though.

Reply:
I think since I'm only calling it once it's not super important, yeah.